src/orderly/cli/__init__.py

```diff
@@ -0,0 +1,204 @@
+import sys
+from collections import Counter
+
+import click
+
+import outpack.search
+from orderly.cli.options import (
+    NumericalParamType,
+    PropagatingGroup,
+    with_root,
+    with_search_options,
+)
+from orderly.run import orderly_run
+from outpack.init import outpack_init
+from outpack.location import (
+    outpack_location_add,
+    outpack_location_list,
+    outpack_location_remove,
+    outpack_location_rename,
+)
+from outpack.util import format_list
+
+
+@click.group(cls=PropagatingGroup)
+@with_root(propagate=True)
+def cli():
+    pass
+
+
+@cli.command()
+@click.argument("path")
+@click.option(
+    "--archive", help="Path to the archive in which packets are stored."
+)
+@click.option(
+    "--use-file-store",
+    is_flag=True,
+    help="Store packets in a content-addressed store.",
+)
+@click.option(
+    "--require-complete-tree",
+    is_flag=True,
+    help=(
+        "Require a complete tree. Whenever packets are pulled from remote"
+        " locations, orderly will ensure all of that packet's dependencies are"
+        " pulled as well."
+    ),
+)
+def init(path, archive, use_file_store, require_complete_tree):
+    """Initialize at new orderly repository in PATH."""
+    kwargs = {}
+    if archive is not None:
+        kwargs["path_archive"] = archive
+    elif use_file_store:
+        kwargs["path_archive"] = None
+    else:
+        # In this case we let outpack_init set the default value for the archive
+        # path, ie. `archive`.
+        pass
+
+    outpack_init(
+        path,
+        use_file_store=use_file_store,
+        require_complete_tree=require_complete_tree,
+        **kwargs,
+    )
+
+
+@cli.command()
+@click.argument("name")
+@click.option(
```

Comment:
There's a balance here to be struck. What has worked for us in the past for this issue is to run the received parameters through a yaml parser, which does a pretty good job usually. Then you accept everything as strings.  Quotes on command lines are always a special sort of pain though

Reply:
I personally have a pretty strong distaste for yaml's interpretation of literals, especially when things like `y` or `off` get converted to booleans.

Alternatively we can reduce the number of flags from 3 to 2 by saying one is for strings only and the other one uses a json/python parser for literals, and can therefore handle booleans, numbers and strings if the user escapes the quotes.

This is what Nix does in a similar context: https://nixos.org/manual/nix/stable/command-ref/nix-shell#opt-arg `--arg` accepts an arbitrary expression, `--argstr` turns the argument into a string literal.  One the one hand I find this quite elegant, and I like having fewer flags, but I would find it more complicated to explain to a lay user than simply having one flag per type.

There's also the option of doing `-p foo int:1234`, which is how I did it in a previous life: https://github.com/microsoft/scitt-ccf-ledger/blob/fd811db702cea568cb3e447ee8e92aa1a57aea02/pyscitt/pyscitt/cli/sign_claims.py#L196-L208

I can't really think of other examples of programs off the top of my head that deal with this issue, although I'm sure they must exist.

Reply:
Strong agree on some of yaml's choices - we usually restrict that a bunch: https://github.com/mrc-ide/orderly2/blob/main/R/util.R#L182-L189 - I don't think anyone has ever written a key `n` and expected that to map to `FALSE`